pkg/apis/core/validation/validation.go

```diff
@@ -3408,6 +3408,11 @@ func ValidatePodSecurityContext(securityContext *core.PodSecurityContext, spec *
 				allErrs = append(allErrs, field.Invalid(fldPath.Child("shareProcessNamespace"), *securityContext.ShareProcessNamespace, "ShareProcessNamespace and HostPID cannot both be enabled"))
 			}
 		}
+		if securityContext.HostUserNamespace != nil {
+			if !utilfeature.DefaultFeatureGate.Enabled(features.NodeUserNamespace) {
```

Comment:
is there a reason the feature gate doesn't match the pod spec field? NodeUserNamespace vs HostUserNamespace?

Reply:
`NodeUserNamespace` feature gate represents user-namespace remapping feature enabled at "Node" level.
While `HostUserNamespace` field in pod spec is a knob/mechanism to escape user-namesapce remapping if "NodeUserNamespace" is enabled. If we name feature gate to match with `HostUserNamespace` that will be less intuitive.